runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -153,15 +153,27 @@ public void updateExecutionStatus(Long stateMachineId,Long taskId, Status status
      * @param stateId
      */
     public void unsidelineState(Long stateMachineId, Long stateId) {
-        State state = this.statesDAO.findById(stateId);
+        State askedState = null;
+        StateMachine stateMachine = retrieveStateMachine(stateMachineId);
+        for (State state: stateMachine.getStates()){
```

Comment:
this will throw null pointer if stateMachine not found. Keep the null check before.